lib/benchee/benchmark/runner.ex

```diff
@@ -29,6 +29,18 @@ defmodule Benchee.Benchmark.Runner do
     Enum.map(scenarios, fn scenario -> parallel_benchmark(scenario, scenario_context) end)
   end
 
+  # This will run the given scenario exactly once, including the before and
+  # after hooks, to ensure the function can execute without raising an error.
+  defp pre_check(scenario, scenario_context = %ScenarioContext{config: %{pre_check: true}}) do
+    scenario_input = run_before_scenario(scenario, scenario_context)
+    scenario_context = %ScenarioContext{scenario_context | scenario_input: scenario_input}
+    _ = measure_iteration(scenario, scenario_context)
+    _ = run_after_scenario(scenario, scenario_context)
+    nil
+  end
+
+  defp pre_check(_, _), do: nil
```

Comment:
`pre_check` is called before `parallel_benchmark` so I like to keep that order :)